File: android/app/src/main/java/investments/hoelzl/tennisace/data/MatchRepository.kt
package investments.hoelzl.tennisace.data

import investments.hoelzl.tennisace.model.Match

/**
 * Abstraction over the match data source. UI and ViewModels depend only on this
 * interface — see docs/API_CONTRACT.md. Today: [MockMatchRepository];
 * later: a Supabase-backed implementation with the same shape.
 */
interface MatchRepository {

    /** All matches, newest first. */
    suspend fun matches(): List<Match>

    /** A single match by id, or null if not found. */
    suspend fun match(id: String): Match?

    /** Persist a new match; returns the stored match (with assigned id). */
    suspend fun add(match: Match): Match
}

File: ios/TennisAce/Services/MatchRepository.swift
import Foundation

/// Abstraction over the match data source. Views and ViewModels depend only on
/// this protocol — see docs/API_CONTRACT.md. Today: `MockMatchRepository`;
/// later: a Supabase-backed implementation with the same shape.
protocol MatchRepository: Sendable {
    /// All matches, newest first.
    func matches() async throws -> [Match]

    /// A single match by id, or nil if not found.
    func match(id: String) async throws -> Match?

    /// Persist a new match; returns the stored match (with assigned id).
    func add(_ match: Match) async throws -> Match
}
